### PR TITLE
Bug 1408912 - For interval based comparison, use time from when new push was sent

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

File: ui/partials/perf/comparechooserctrl.html [hunk 1]
```diff
@@ -13,7 +13,7 @@
             <label><input type="checkbox" ng-model="revisionComparison">Compare with a specific revision</label>
           </div>
           <div ng-show="!revisionComparison">
-            <p class="help-block">By default, Perherder will compare against performance data gathered over the last 2 days</p>
+            <p class="help-block">By default, Perherder will compare against performance data gathered over the last 2 days from when new revision was pushed</p>
```
Comment (rwood-moz) [inline ui/partials/perf/comparechooserctrl.html]: Nit: there's a legacy typo 'Perherder' (although with the loading cat gif maybe purrherder would work hah)

File: ui/js/services/perf/compare.js [hunk 2]
```diff
@@ -217,62 +217,52 @@ treeherder.factory('PhCompare', [
...
-                                framework: _.uniq(_.map(seriesChunk, 'frameworkId')),
-                                push_id: params.pushIDs }
-                        ).then(function (seriesData) {
-                            params.pushIDs.forEach(function (resultSetId) {
-                                if (resultsMap[resultSetId] === undefined) {
-                                    resultsMap[resultSetId] = {};
+            getResultsMap: (projectName, seriesList, params) => {
+                let resultsMap = {};
+                return $q.all(_.chunk(seriesList, 20).map(
+                    seriesChunk => PhSeries.getSeriesData(
```
Comment (rwood-moz) [inline ui/js/services/perf/compare.js]: Previously, before calling PhSeries.getSeriesData, params.pushIDs was verified to exist, would that cause a problem not checking that first?

File: ui/js/services/perf/compare.js [hunk 3]
```diff
@@ -217,62 +217,52 @@ treeherder.factory('PhCompare', [
...
-                                    frameworkId: series.frameworkId,
-                                    values: _.map(data, 'value')
-                                };
-                            });
-                        });
-                })).then(function () {
-                    return resultsMap;
-                });
+                        })
+                )).then(() => resultsMap);
```
Comment (rwood-moz) [inline ui/js/services/perf/compare.js]: Oh cool so you don't even need a 'return' before resultsMap (?)

File: ui/partials/perf/comparechooserctrl.html (hunk 1 above)
Comment (wlach) [reply ui/partials/perf/comparechooserctrl.html]: lol, I will fix this as a quick followup

File: ui/js/services/perf/compare.js (hunk 3 above)
Comment (wlach) [reply ui/js/services/perf/compare.js]: Nope! Return is assumed if the function is a single expression. :)

File: ui/js/services/perf/compare.js (hunk 2 above)
Comment (wlach) [reply ui/js/services/perf/compare.js]: That check was part of an if/else construction which I managed to eliminate by passing the parameters directly into the function (and then from there into the api request). So pretty sure this is ok. :)

**Pull request merged**